app/controllers/api/storage.php

```diff
@@ -752,32 +873,50 @@
             $contentType = $file->getAttribute('mimeType');
         }
 
-        $source = $device->read($path);
-
-        if (!empty($file->getAttribute('openSSLCipher'))) { // Decrypt
-            $source = OpenSSL::decrypt(
-                $source,
-                $file->getAttribute('openSSLCipher'),
-                App::getEnv('_APP_OPENSSL_KEY_V'.$file->getAttribute('openSSLVersion')),
-                0,
-                \hex2bin($file->getAttribute('openSSLIV')),
-                \hex2bin($file->getAttribute('openSSLTag'))
-            );
-        }
-
-        $output = $compressor->decompress($source);
         $fileName = $file->getAttribute('name', '');
 
-        // Response
-        $response
-            ->setContentType($contentType)
-            ->addHeader('Content-Security-Policy', 'script-src none;')
-            ->addHeader('X-Content-Type-Options', 'nosniff')
-            ->addHeader('Content-Disposition', 'inline; filename="'.$fileName.'"')
-            ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + (60 * 60 * 24 * 45)).' GMT') // 45 days cache
-            ->addHeader('X-Peak', \memory_get_peak_usage())
-            ->send($output)
-        ;
+        if ($device->getFileSize($path) > APP_LIMIT_COMPRESSION) {          
+            $response
+                ->setContentType($contentType)
+                ->addHeader('Content-Security-Policy', 'script-src none;')
+                ->addHeader('X-Content-Type-Options', 'nosniff')
+                ->addHeader('Content-Disposition', 'inline; filename="' . $fileName . '"')
+                ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + (60 * 60 * 24 * 45)) . ' GMT') // 45 days cache
+                ->addHeader('X-Peak', \memory_get_peak_usage())
+                ->addHeader('Content-Length',$device->getFileSize($path))
+            ;
+            $handle = fopen($path, 'rb');
+            while(!feof($handle)) {
+                $response->chunk(@fread($handle,APP_CHUNK_SIZE), feof($handle));
+            }
+            fclose($handle);
```

Comment:
You can return void here, (`return;`) and skip the big if<>else wrapper that makes the code a bit harder to read.

Reply:
done!